SignalServiceKit/src/Messages/OWSMessageUtils.m

```diff
@@ -149,11 +151,46 @@ + (nullable TSQuotedMessage *)quotedMessageForMessage:(TSMessage *)message
         TSAttachment *_Nullable attachment =
             [TSAttachment fetchObjectWithUniqueID:attachmentId transaction:transaction];
         if (attachment) {
-            sourceFilename = attachment.sourceFilename;
-            contentType = attachment.contentType;
-            // Try to generate a thumbnail, if possible.
-            thumbnailData = [self thumbnailDataForAttachment:attachment];
-            hasAttachment = YES;
+            // If the attachment is "oversize text", try to treat it appropriately.
+            if (!hasText && [NSObject isNullableObject:attachment.contentType equalTo:OWSMimeTypeOversizeTextMessage] &&
+                [attachment isKindOfClass:[TSAttachmentStream class]]) {
+
+                hasText = YES;
+                quotedText = @"";
+
+                TSAttachmentStream *attachmentStream = (TSAttachmentStream *)attachment;
+                NSData *_Nullable oversizeTextData = [NSData dataWithContentsOfFile:attachmentStream.filePath];
+                if (oversizeTextData) {
+                    NSString *_Nullable oversizeText =
+                        [[NSString alloc] initWithData:oversizeTextData encoding:NSUTF8StringEncoding];
+                    // First, truncate to the rough max characters.
+                    NSString *_Nullable truncatedText =
+                        [oversizeText substringToIndex:kOversizeTextMessageSizeThreshold - 1];
```

Comment:
So the usage of `kOversizeTextMessageSizeThreshold` here is pretty arbitrary, it's just some value which is "much more" than three lines, so that when we display the three lines of text we can be sure there's enough?

Reply:
It's only arbitrary in the sense that `kOversizeTextMessageSizeThreshold` is out limit on how long text should be in protos since they'll be stored in the database.  Similarly, the idea here is to cap the length of the "text snippet" we include in the "quoted reply" proto.